### ISSUE TITLE
[BUG] Select values in templatefields are returning their indexes, not values

### ISSUE THREAD

**Issue by Maelstromeous** [bolt/bolt#6325]
It appears that values in templateselect fields are returning their index of the selected value, not the actual value itself.

In `theme.yml`:

```
services/fire-extinguishers.twig:
    ticks:
        type: repeater
        label: "Tickmarks"
        fields:
            kind:
                type: select
                values: [blue, grey]
                required: true
            text:
                type: text
                required: true
```

So selecting `blue` in the backend will result in `0` when using:

```
{% for tick in record.templatefields.ticks %}
    {{ tick.get('kind') }}
{% endfor %}
```

Selecting `grey` would also result in `1`, rather than the value.

Object generated:

![object](https://cloud.githubusercontent.com/assets/1776058/22422370/e0181f30-e6e3-11e6-8cbb-41a2a17347c3.png)

Details
-------

 - Relevant Bolt Version: 3.2.5
 - Install type: Composer install
 - PHP version: 5.6
 - Used web server: Apache

Reproduction
------------

​1. Create a record and configure `theme.yml` with the above config.
2. In the record, create an entry for the repeater and select a value
3. Create a 2nd entry and choose another value
4. Observe the record and it's contents, showing a `0` and a `1` respectively as their values

**Comment by rossriley** [comment]
Found the problem with this, working on fix.

**Comment by CarsonF** [comment]
@rossriley figure it out?

**Comment by GawainLynch** [comment]
@rossriley Any update on this (and by proxy #6336)

**Comment by rossriley** [comment]
Yes, I need to pull my finger out.... PR coming in the next couple of hours.

**Comment by GawainLynch** [comment]
Fixed in #6405
